octavia-cli/integration_tests/test_apply/test_resources.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import os
+
+import pytest
+import yaml
+from octavia_cli.apply.resources import Connection, Destination, Source
+from octavia_cli.entrypoint import get_api_client, get_workspace_id
+
+pytestmark = pytest.mark.integration
+
+
+@pytest.fixture(scope="module")
+def api_client():
+    return get_api_client("http://localhost:8000")
+
+
+@pytest.fixture(scope="module")
+def workspace_id(api_client):
+    return get_workspace_id(api_client, None)
+
+
+@pytest.fixture(scope="module")
+def source_configuration_and_path():
+    path = f"{os.path.dirname(__file__)}/configurations/sources/poke/configuration.yaml"
+    with open(path, "r") as f:
+        local_configuration = yaml.safe_load(f)
+    return local_configuration, path
+
+
+def silent_remove(path):
+    try:
+        os.remove(path)
+        return True
+    except FileNotFoundError:
+        return False
+
+
+@pytest.fixture(scope="module")
+def source_state_path():
+    state_path = f"{os.path.dirname(__file__)}/configurations/sources/poke/state.yaml"
+    silent_remove(state_path)
```

Comment:
I don't really understand why these methods are calling `remove` on `state_path`? Or why its calling `silent_remove()` twice?

And similarly, why are the `source()` and `destination()` methods calling `delete_source()`/`delete_destination()` after yielding them?

Reply:
In the context of pytest fixture, the code after yielding is considered as the test teardown logic. I'm basically cleaning up state files and created remote ressource at the end of the test run.